httpclient/useragent.go

```diff
@@ -38,3 +40,134 @@ func (rt *userAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 	return rt.inner.RoundTrip(req)
 }
+
+func UserAgent(version string) *userAgent {
```

Comment:
Yeah, it came to my mind too - thanks for raising it!

The motivation here was to kinda force people to use a function, so that we get the chance to source the ENV variable, which would otherwise be difficult when providers just build exported struct, and/or awkward if we plugged this logic into `String()`.